policy-controller/k8s/index/src/outbound/index.rs

```diff
@@ -229,32 +239,43 @@ impl Namespace {
                 continue;
             }
 
-            if let Some(port) = parent_ref.port {
-                if let Some(port) = NonZeroU16::new(port) {
-                    let service_port = ServicePort {
-                        port,
-                        service: parent_ref.name.clone(),
-                    };
-                    tracing::debug!(
-                        ?service_port,
-                        route = route.name(),
-                        "inserting route for service"
-                    );
-                    let service_routes =
-                        self.service_routes_or_default(service_port, cluster_info, service_info);
-                    service_routes.apply(route.gkn(), outbound_route.clone());
-                } else {
-                    tracing::warn!(?parent_ref, "ignoring parent_ref with port 0");
-                }
+            let port = parent_ref.port.and_then(NonZeroU16::new);
+            if let Some(port) = port {
+                let service_port = ServicePort {
+                    port,
+                    service: parent_ref.name.clone(),
+                };
+                tracing::debug!(
+                    ?service_port,
+                    route = route.name(),
+                    "inserting route for service"
+                );
+                let service_routes =
+                    self.service_routes_or_default(service_port, cluster_info, service_info);
+                service_routes.apply(route.gkn(), outbound_route.clone());
             } else {
-                tracing::warn!(?parent_ref, "ignoring parent_ref without port");
+                // If the parent_ref doesn't include a port, apply this route
+                // to all ServiceRoutes which match the Service name.
+                self.service_port_routes.iter_mut().for_each(
+                    |(ServicePort { service, port: _ }, routes)| {
+                        if service == &parent_ref.name {
+                            routes.apply(route.gkn(), outbound_route.clone());
+                        }
+                    },
+                );
```

Comment:
This is an O(N) scan of all routes in a namespace. That's probably acceptable in the short term, but it seems easily avoidable if we change the structure of service_port_routes so that we can lookup by service.

Reply:
Ah this isn't actually iterating over all routes. all service:ports. That's probably more reasonable.